### PR TITLE
Introduce ValidatorApiChannel and use it for validator duties

### PR DESCRIPTION
**Pull request by ajsutton**
## PR Description
Introduce an event channel to provide validator API data, beginning with validator duties. Switch REST API over to using it.

### CODE REVIEW

File: services/beaconchain/src/main/java/tech/pegasys/artemis/services/beaconchain/BeaconChainController.java [hunk 1]
```diff
@@ -94,6 +96,7 @@
   private volatile SyncService syncService;
   private volatile AttestationManager attestationManager;
   private volatile ValidatorCoordinator validatorCoordinator;
+  private CombinedChainDataClient combinedChainDataClient;
```
Comment (CjHare) [inline services/beaconchain/src/main/java/tech/pegasys/artemis/services/beaconchain/BeaconChainController.java]: There is no `volatile` keyword on the `combinedChainDataClient`, which is inconsistent with the other fields, is that intentional?

File: services/beaconchain/src/main/java/tech/pegasys/artemis/services/beaconchain/BeaconChainController.java [hunk 2]
```diff
@@ -166,9 +170,16 @@ public void initAll() {
     initAttestationPropagationManager();
     initP2PNetwork();
     initSyncManager();
+    initValidatorApiHandler();
+    initValidatorCoordinator();
     initRestAPI();
   }
 
+  private void initCombinedChainDataClient() {
+    HistoricalChainData historicalChainData = new HistoricalChainData(eventBus);
```
Comment (macfarla) [inline services/beaconchain/src/main/java/tech/pegasys/artemis/services/beaconchain/BeaconChainController.java:180]: do we want a LOG.debug here?

**Review by macfarla — APPROVED** [review]
LGTM

File: validator/api/src/main/java/tech/pegasys/artemis/validator/api/ValidatorDuties.java [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+public class ValidatorDuties {
+  private final BLSPublicKey publicKey;
+  private final Optional<Duties> duties;
+
+  private ValidatorDuties(final BLSPublicKey publicKey, final Optional<Duties> duties) {
+    this.publicKey = publicKey;
+    this.duties = duties;
+  }
+
+  public static ValidatorDuties forKnownValidator(
```
Comment (cemozerr) [inline validator/api/src/main/java/tech/pegasys/artemis/validator/api/ValidatorDuties.java]: nit: 
```suggestion
  public static ValidatorDuties forValidatorWithKnownDuties(
```

Comment (ajsutton) [reply validator/api/src/main/java/tech/pegasys/artemis/validator/api/ValidatorDuties.java]: Wound up changing both so we have `noDuties` and `withDuties`.

File: services/beaconchain/src/main/java/tech/pegasys/artemis/services/beaconchain/BeaconChainController.java (hunk 2 above)
Comment (ajsutton) [reply services/beaconchain/src/main/java/tech/pegasys/artemis/services/beaconchain/BeaconChainController.java:180]: Done, though I don't really know why we have all those debug statements... Probably should make them all trace or get rid of them at some point.

File: services/beaconchain/src/main/java/tech/pegasys/artemis/services/beaconchain/BeaconChainController.java (hunk 1 above)
Comment (ajsutton) [reply services/beaconchain/src/main/java/tech/pegasys/artemis/services/beaconchain/BeaconChainController.java]: Good spot.  Kind of annoying that we need volatile there and would be good to refactor so everything is created by a builder and injected to the constructor, but would have to separate construction from starting properly across all those classes first.

**Pull request merged**